render/egl.c

```diff
@@ -342,7 +351,20 @@ bool wlr_egl_swap_buffers(struct wlr_egl *egl, EGLSurface surface,
 		int nrects;
 		pixman_box32_t *rects =
 			pixman_region32_rectangles(&flipped_damage, &nrects);
-		EGLint egl_damage[4 * nrects];
+
+		static EGLint *egl_damage;
+		static uint32_t egl_damage_size = 0;
+
+		// Expand egl_damage if it isn't large enough.
+		if (egl_damage_size < (uint32_t) 4 * nrects) {
+			egl_damage = malloc(sizeof(EGLint) * (4 * nrects));
```

Comment:
I don't see where this is freed

Reply:
I've made `egl_damage` static since we want to keep some allocated memory quickly available for this function. It won't be freed until the programs exits, I guess.

Reply:
I see. But you still allocate a new one if the size grows, so you need to free the previous one there when allocating a new one at least.